### PR TITLE
created lti monitor parameters service

### PR DESCRIPTION
**Pull request by zagovorichev**
Pull request summary
 
https://oat-sa.atlassian.net/browse/ACTP-221
 
Added service to make monitoring params configurable

#### How to test
 
Proctoring page should work as expected (as worked before)

#### Dependencies
Companion PR :
 - [ ] https://github.com/oat-sa/extension-tao-act/pull/1297
 - [ ] https://github.com/oat-sa/extension-tao-proctoring/pull/768

### CODE REVIEW

File: model/LtiMonitorParametersService.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    /**
+     * @return array
+     * @throws LtiException
+     * @throws LtiVariableMissingException
+     * @throws common_exception_Error
+     */
+    public function getParameters()
+    {
+        $params = [];
+        $params['defaultTag'] = $this->getDefaultTag();
```
Comment (shpran) [inline model/LtiMonitorParametersService.php]: Nip: can be replaced with
```
$params = [
    'defaultTag' => $this->getDefaultTag(),
];
```

File: model/LtiMonitorParametersService.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            : '';
+    }
+
+    /**
+     * @return LtiLaunchDataService
+     */
+    private function getLtiLaunchDataService()
+    {
+        /** @var LtiLaunchDataService $service */
+        $service = $this->getServiceLocator()->get(LtiLaunchDataService::SERVICE_ID);
```
[the remark this answers is not in the document]
Comment (zagovorichev) [reply model/LtiMonitorParametersService.php]: I know, but my editor highlights that like a warning

File: controller/Monitor.php [hunk 3]
```diff
@@ -18,30 +19,33 @@
...
-        $params = [
-            'defaultTag' => (string)$this->getDefaultTag(),
-        ];
-
-        if (!is_null($delivery)) {
-            $params['delivery'] = $delivery->getUri();
-        }
-        $this->setClientRoute(_url('index', 'Monitor', 'taoProctoring', $params));
+        /** @var LtiMonitorParametersService $monitoringParamsService */
+        $monitoringParamsService = $this->getServiceLocator()->get(LtiMonitorParametersService::class);
```
[the remark this answers is not in the document]
Comment (zagovorichev) [reply controller/Monitor.php]: yep, you're right

File: model/LtiMonitorParametersService.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    private $launchData;
+
+    /**
+     * @return array
+     * @throws LtiException
+     * @throws LtiVariableMissingException
+     * @throws common_exception_Error
+     */
+    public function getParameters()
```
[the remark this answers is not in the document]
Comment (zagovorichev) [reply model/LtiMonitorParametersService.php]: done

**Pull request merged**